core/src/main/java/org/apache/iceberg/BaseIncrementalChangelogScan.java

```diff
@@ -72,13 +72,16 @@ protected CloseableIterable<ChangelogScanTask> doPlanFiles(
             .filter(manifest -> changelogSnapshotIds.contains(manifest.snapshotId()))
             .toSet();
 
+    scanMetrics().totalDataManifests().increment(newDataManifests.size());
```

Comment:
I don't think we support delete manifests atm for changelog/append scans, so this is only counting data manifests here and above

Reply:
Support for deletes is being added by https://github.com/apache/iceberg/pull/6182

Reply:
I agree. Since this is only operating on data manifests right now, we shouldn't need to count delete manifests yet.